### ISSUE TITLE
[@types/ember-data bug] - 3.1.0 registries moving breaks ember-decorators

### ISSUE THREAD

**Issue by csantero** [typed-ember/ember-cli-typescript#329]
### Which package(s) does this problem pertain to?
  - [x] @types/ember-data

### What are instructions we can follow to reproduce the issue?
```sh
ember new sample; cd ./sample # Create a new ember app
ember install ember-cli-typescript # Set up typescript support
ember install ember-decorators
rm types/ember-data.d.ts
```

Add the following file at `app/models/person.ts`:

```ts
import { attr } from '@ember-decorators/data';
import DS from 'ember-data';

export default class Person extends DS.Model {
  @attr('string') firstName!: string;
}

declare module 'ember-data/types/registries/model' {
  export default interface ModelRegistry {
    'person': Person;
  }
}
```
ember build

### Now about that bug. What did you expect to see?
No errors

### What happened instead?

```bash
node_modules/@ember-decorators/data/index.d.ts(1,14): error TS2305: Module '"/home/chris/projects/sample/node_modules/@types/ember-data/index"' has no exported member 'ModelRegistry'.
node_modules/@ember-decorators/data/index.d.ts(1,29): error TS2305: Module '"/home/chris/projects/sample/node_modules/@types/ember-data/index"' has no exported member 'TransformRegistry'.
node_modules/@ember-decorators/data/index.d.ts(1,14): error TS2305: Module '"/home/chris/projects/sample/node_modules/@types/ember-data/index"' has no exported member 'ModelRegistry'.
node_modules/@ember-decorators/data/index.d.ts(1,29): error TS2305: Module '"/home/chris/projects/sample/node_modules/@types/ember-data/index"' has no exported member 'TransformRegistry'.
```

Relevant deps:
```
    "@types/ember": "^3.0.24",
    "@types/ember-data": "^3.1.1",
    "@types/ember-qunit": "^3.4.2",
    "@types/ember-test-helpers": "^1.0.3",
    "@types/ember-testing-helpers": "^0.0.3",
    "@types/ember__test-helpers": "^0.7.5",
    "@types/qunit": "^2.5.3",
    "@types/rsvp": "^4.0.2",
    "typescript": "^3.1.1"
```

**Comment by mike-north** [comment]
fixes are taking place in ember-decorators

**Comment by csantero** [comment]
I can confirm this is fixed with ember-decorators 2.5.2
